modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/PlannerHelper.java

```diff
@@ -180,7 +180,7 @@ public static IgniteRel optimize(SqlNode sqlNode, IgnitePlanner planner) {
                 LOG.debug(planner.dump());
             }
 
-            if (ex instanceof CannotPlanException) {
+            if (ex instanceof CannotPlanException || ex instanceof SqlException) {
                 throw ex;
             } else if (ex.getClass() == RuntimeException.class && ex.getCause() instanceof SqlException) {
```

Comment:
but now the code is dead, isn't it?

Reply:
No, it's a different case.
Test `ForceIndexHintPlannerTest.testWithMultipleIndexHints` reproduces it